### DISCUSSION TITLE
skip pinning for specific dependencies

### DISCUSSION THREAD

**Discussion by wyardley** [renovatebot/renovate#15934] [Request Help]
I typically follow Renovate's default of pinning our library dependencies, however, there are times where I want to pin something (a CircleCI orb, maybe) to a major dep only (e.g., `foo@1`), and leave it that way

I resolved this one time by closing the PR to pin that dep, however, that resulted in the branch `renovate/pin-dependencies` being marked as closed / blocked completely.

I know I could probably customize the pin behavior for certain types of deps, but is there any middle ground (without getting too crazy with the deps) that will let me ignore pins on specific deps without fully suppressing the overall behavior of pinning deps?

**Comment by JamieMagee** [comment]
It sounds like you want to set `packageRules` to customize the `rangeStrategy`[^1] for a specific set of dependencies? If I'm understanding you correctly you're extending `config:base` which sets `rangeStrategy` to `auto`. So something like this might work for you:

```json
"packageRules": [
  {
    "matchPackageNames": ["..."],
    "rangeStrategy": "in-range-only" // Update the lock file when in-range updates are available, ignore package file updates
  }
],
```

[^1]: https://docs.renovatebot.com/configuration-options/#rangestrategy

**Reply by sheldonhull** [reply]
Hmm, I'll give this a try. I thought I tried in-range-only and it didn't seem to work, but I'll try again and confirm! thank you!

**Reply by sheldonhull** [reply]
Ok so that misses what I'm trying to do:

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "description": "these are rules to always disable regardless, to avoid edge cases, or problematic updates",
  "packageRules": [
    {
      "description": "bypass pinning terraform resuable module provider versions. Terraform files in the root are resuable modules, not 'root modules', using cloudposse module project structure. Therefore these should _not_ do version pinning on providers, but instead just honor a minimum range to avoid failure with pinned root modules calling a module with a different version, which would cause an error. Examples nested directory should still version pin.",
      "enabled": true,
      "matchFileNames": ["versions.tf"],
      "matchDatasources": ["terraform-provider"],
      "rangeStrategy": "in-range-only"
    }
  ]
}


```

Essentially `{reporoot}/*.tf` though right now I'm just try fixed versions.tf should honor and leave >= 3.1.0 alone. The pinning shouldn't happen. Right now with `in-range-only` it doesn't work and still tries to pin.

**Comment by wyardley** [comment]
Well, in my case, I'd use a totally different source, and that might work.

However, in the past, I'd seemed to be able to dismiss certain pin PRs manually, and have those not come back, without blocking pinning _any_ PR; in this case, though, it seems like Renovate is using the same branch for all pins, so I believe this may not be possible?

Oddly, I have _another_ CircleCI orb that's also not pinned (i.e., it's just `@N`) and the PR is _not_ trying to adjust that one.

**Comment by sheldonhull** [comment]
Ok I think this is working so far...

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": ["local>devex/renovate-config"],
  "packageRules": [
    {
      "description": "bypass pinning terraform resuable module provider versions. Terraform files in the root are resuable modules, not 'root modules', using cloudposse module project structure. Therefore these should _not_ do version pinning on providers, but instead just honor a minimum range to avoid failure with pinned root modules calling a module with a different version, which would cause an error. Examples nested directory should still version pin.",
      "enabled": false,
      "matchFileNames": ["*.tf"],
      "matchDatasources": ["terraform-provider"],
      "matchUpdateTypes": ["pin"] // 👈👈👈👈👈👈 this is what was missing, otherwise pinning seemed to happen regardless.
    }
  ]
}
```

Match the update type and the specific root path without glob and it still worked for examples directory and removed changes to root of repo.
